src/languages/bigquery/bigquery.formatter.ts

```diff
@@ -20,122 +21,58 @@ const reservedCommands = [
   'WITH',
   'OMIT RECORD IF', // legacy
   // DML, https://cloud.google.com/bigquery/docs/reference/standard-sql/dml-syntax
-  'INSERT',
-  'INSERT INTO',
+  'INSERT [INTO]',
   'VALUES',
-  'DELETE',
-  'DELETE FROM',
+  'DELETE [FROM]',
   'TRUNCATE TABLE',
   'UPDATE',
-  'MERGE',
-  'MERGE INTO',
+  'MERGE [INTO]',
   // 'USING',
   // DDL, https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language
-  'CREATE SCHEMA',
-  'CREATE SCHEMA IF NOT EXISTS',
+  'CREATE SCHEMA [IF NOT EXISTS]',
   'DEFAULT COLLATE',
-  'CREATE TABLE',
-  'CREATE TABLE IF NOT EXISTS',
-  'CREATE TEMP TABLE',
-  'CREATE TEMP TABLE IF NOT EXISTS',
-  'CREATE TEMPORARY TABLE',
-  'CREATE TEMPORARY TABLE IF NOT EXISTS',
+  'CREATE [OR REPLACE] [TEMP|TEMPORARY|SNAPSHOT|EXTERNAL] TABLE [IF NOT EXISTS]',
   'CLUSTER BY',
-  'CREATE OR REPLACE TABLE',
-  'CREATE OR REPLACE TEMP TABLE',
-  'CREATE OR REPLACE TEMPORARY TABLE',
   'FOR SYSTEM_TIME AS OF', // CREATE SNAPSHOT TABLE
-  'CREATE SNAPSHOT TABLE',
-  'CREATE SNAPSHOT TABLE IF NOT EXISTS',
-  'CREATE VIEW',
-  'CREATE VIEW IF NOT EXISTS',
-  'CREATE OR REPLACE VIEW',
-  'CREATE MATERIALIZED VIEW',
-  'CREATE OR REPLACE MATERIALIZED VIEW',
-  'CREATE MATERIALIZED VIEW IF NOT EXISTS',
-  'CREATE EXTERNAL TABLE',
-  'CREATE OR REPLACE EXTERNAL TABLE',
-  'CREATE EXTERNAL TABLE IF NOT EXISTS',
+  'CREATE [OR REPLACE] [MATERIALIZED] VIEW [IF NOT EXISTS]',
   'WITH CONNECTION',
   'WITH PARTITION COLUMNS',
-  'CREATE FUNCTION',
-  'CREATE OR REPLACE FUNCTION',
-  'CREATE FUNCTION IF NOT EXISTS',
-  'CREATE TEMP FUNCTION',
-  'CREATE OR REPLACE TEMP FUNCTION',
-  'CREATE TEMP FUNCTION IF NOT EXISTS',
-  'CREATE TEMPORARY FUNCTION',
-  'CREATE OR REPLACE TEMPORARY FUNCTION',
-  'CREATE TEMPORARY FUNCTION IF NOT EXISTS',
+  'CREATE [OR REPLACE] [TEMP|TEMPORARY|TABLE] FUNCTION [IF NOT EXISTS]',
```

Comment:
This will also generate `CREATE OR REPLACE FUNCTION IF NOT EXISTS`, but `OR REPLACE` and `IF NOT EXISTS` can't co-exist.

Reply:
Yep, I know. As mentioned in PR description.

It doesn't really matter a lot that the formatter is capable of formatting some incorrect syntax. I'm more concerned about memory overhead from generating these extra impossible variants. But I think that's a minuscule loss compared to the win in readability.